Element84/filmdrop-ui
# Allow defining collections via config

**Related Issue(s):**

- NA

**Proposed Changes:**

1. Adds a `Collections` option in config.
2. Adds error handling if no collections found or matched.

Bonus:
- Fix to make `DEFAULT_COLLECTION` from config actually optional and not error out if not in API collections returned.
- Resolves linting error for `label` in `ViewSelector.jsx` not being connected to a element.

### To Test:

- `npm run test` 
- confirm automated test pass 🍏 

**Tests for Collections in config**
- make sure `COLLECTIONS` not added to config.json
- make sure `DEFAULT_COLLECTION` is not included in config.json
- set `STAC_API_URL` to be using an API from earth search v1
- load app
- confirm all collections are shown in dropdown
- run some searches with collections
- confirm search works as expected
- stop app
- add `"COLLECTIONS": ["naip", "sentinel-2-l2a"]` to config.json
- load app
- confirm dropdown shows only `naip` and `sentinel-2-l2a` in list 
- confirm dropdown shows `naip` first (should be selected initially)
- run some searches with collections
- confirm search works as expected
- stop app
- add `"DEFAULT_COLLECTION": "sentinel-2-l2a"` to config.json
- load app
- confirm dropdown shows only `naip` and `sentinel-2-l2a` in list 
- confirm dropdown shows `sentinel-2-l2a` first (should be selected initially)
- stop app
- change `COLLECTIONS` to not match any of the STAC API collections, `"COLLECTIONS": [testCollection]` in config.json
- load app
- confirm error message loads
- confirm error message shows `Error: No Collections Found`

**PR Checklist:**
